### PR TITLE
Block Navigation: try using CSS for indentation with known max indent instead of spacer divs

### PR DESCRIPTION
**Pull request by gwwar**
In the Block Navigation component, I noticed that we were using spacer divs instead of CSS rules. For a deeply nested child, we'd see X divs for the necessary indentation at that level.

Some blocks allow unbounded nesting, like the columns block. Proposed changes here would remove the need for an excessive number of whitespace divs. I've arbitrarily picked 8 as the max indentation level, but I'd be open to other limits/approaches.

What's a little frustrating is that [attr()](https://developer.mozilla.org/en-US/docs/Web/CSS/attr()) doesn't have support outside of content, but maybe in the future this can be done simply as a calc + attr rule.

| before | after |
|-----|-----|
| Screen Shot 2021-05-20 at 9 33 15 AM | Screen Shot 2021-05-20 at 10 21 33 AM |

#### Testing Instructions

* Checkout this branch
* Activate a theme that support full site editing, like TT1 blocks
* Visit the Site Editor at `/wp-admin/admin.php?page=gutenberg-edit-site`
* Click on list view in the top toolbar. (The icon to the right of the redo button).
* Try adding heavily nested blocks, like column or group. Add enough so we go past 9 levels.
* Verify that there are no regressions for levels up to 9 deep. Test and see if having a max indent level still feels usable. (Previously we could grow the size of the left panel arbitrarily large, making the content pane in the center shrink)

### CODE REVIEW

File: packages/block-editor/src/components/block-navigation/style.scss [hunk 1]
```diff
@@ -274,7 +274,34 @@
 	}
 }
 
-.block-editor-block-navigator-indentation {
-	flex-shrink: 0;
-	width: $grid-unit-30 + $grid-unit-05; // Indent a full icon size, plus 4px which optically aligns child icons to the text label above.
+// First level of indentation is aria-level 2, max indent is 8.
```
Comment (jasmussen) [inline packages/block-editor/src/components/block-navigation/style.scss:277]: Should we use a SCSS for loop for this one?

**Review by jasmussen — APPROVED** [review]
Love it a ton. Before:

before

After:

Screenshot 2021-05-21 at 09 21 16

Much better!

**Comment by talldan** [comment]
This is a nice simplification. I feel like there could still be some indication that flattened deeply nested blocks are actually nested and not siblings, but that could always be a follow-up if a designer has an idea.

**Comment by jasmussen** [comment]
>  I feel like there could still be some indication that flattened deeply nested blocks are actually nested and not siblings

Yes, that's a good point. 

I think it's okay to ship as is as more than 8 levels of nesting really should be a super rare occurrence, to the point that we might even want to just prepend an ↳ icon or something.

File: packages/block-editor/src/components/block-navigation/style.scss (hunk 1 above)
Comment (gwwar) [reply packages/block-editor/src/components/block-navigation/style.scss:277]: Good call I added this in https://github.com/WordPress/gutenberg/pull/32063/commits/e5c2be34da74e3c029d886308892f97fde3efc1a

Generated final CSS looks like:
Screen Shot 2021-05-21 at 8 57 24 AM

**Comment by gwwar** [comment]
> I feel like there could still be some indication that flattened deeply nested blocks are actually nested and not siblings

I'm looking at adding expand/collapse controls next in https://github.com/WordPress/gutenberg/issues/26141 which I think would cover this case. Happy to follow up on this otherwise in another PR.

**Comment by gwwar** [comment]
Thanks for the reviews @jasmussen @talldan !

**Comment by jasmussen** [comment]
👌

**Pull request merged**